### ISSUE TITLE
Game "pause" constantly and randomly

### ISSUE THREAD

**Issue by cjxgm** [KillaW0lf04/The-Witcher-2-Issues#101]
I'm using ArchLinux x86_64 with xbox wired controller with xboxdrv driver, and I'm using the latest linux_beta build of the game on steam.

The game will "freeze" randomly, but the audio is still playing, looks like it has been paused because of losing window focus. And if I switch to other window and then switch back, it continues to run.

I ran the game fullscreen and is using gnome shell 3.14.3, with bumblebee (primusrun, nvidia), with gl-multithreaded enabled.

**Comment by Dncnsnwdn** [comment]
I think I may be seeing the same issue, except that it's reliably after the first flashback cutscene and I can't get the game to continue to run.

The symptoms are very similar though. The cutscene plays, Roche says, “What just happened?”, and the game freezes on a black screen with 100% CPU usage on one core. Switching windows plays Geralt's next line, but nothing else. Doing it again gives Roche's next line, and the 3D view returns, but static, and that's it. Nothing else advances the game. The background ambient audio continues to play.

I saw the same thing a couple of times before this point (like cjxm's random freezes) but quitting and reloading fixed it.

I'm also on Arch x86_64, with nvidia 346.35 (no bumblebee, though), using Openbox. Also running the game fullscreen (windowed makes the switching trick easier, but doesn't help appreciably). Hardware is Gigabyte M68MT-M2 mobo, AMD FX-4100, Nvidia GT 630, 4Gb RAM.

**Comment by vpltd-peterm** [comment]
That sounds like a deadlock is happening in the nvidia driver. Does it happen without _GL_THREADED_OPTIMIZATIONS enabled?

**Comment by vpltd-peterm** [comment]
Also when it deadlocks, run a "top -H -p <pid of witcher 2>" and tell me what the name of the thread using all the CPU time is ?

**Comment by Dncnsnwdn** [comment]
No, as far as I can tell, _GL_THREADED_OPTIMIZATIONS makes no difference (that was the first thing I thought of, too). And of course I should have thought of checking with top: it's G.14 that's grabbing all the CPU time.

Interestingly, the dialog is currently progressing perfectly on another workspace, and as soon as I switch back it freezes again with G.14 taking 100%. And now I discover that when control has been returned to the player, just before the fistfight with the prison guards, it's fine again. (So at least I'll be able to continue the game. :) )

**Comment by cjxgm** [comment]
Disabling _GL_THREADED_OPTIMIZATIONS works for me though.

**Comment by Dncnsnwdn** [comment]
Yes, I'm beginning to think it's not the same issue at all,
mainly because I've seen something more like what you reported myself
now, once or twice. The game freezes, but a quick switch away and back
again clears it. Mine was much more intrusive.

**Comment by vpltd-peterm** [comment]
I have seen this problem now where the game requires a switch away and back again to clear a frozen state. It looked as if our E.d3d9.GL thread has stopped and the game's rendering thread G.14 is waiting for it.

However, every time I attach GDB to it to find out what's going on, it unfreezes. Grr :)
I also cant reliably reproduce it. I've had it happen twice after hours of playing.

I suspect this is a bug in the nVidia drivers, but can't prove it. I can't replicate it on my AMD system yet, but perhaps i didn't play for long enough yet...

**Comment by Dncnsnwdn** [comment]
Should I open another ticket for "my" bug then, since it seems to be
different (albeit maybe related)? I don't think there's anything more to
report (except that it hasn't recurred on any later flashback
cutscenes), unless there's something else you'd like me to try/check.

**Comment by vpltd-peterm** [comment]
No need. In fact, that is exactly where I saw the problem too, on my Ubuntu 14.10 system using XFCE. 
I will look into this, it is more likely something is stopping our GL worker thread running, and the window losing focus just happens to fix it. I need to be able to reproduce it reliably though :(
